src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -527,10 +552,15 @@ private void AttributePrefix(IEnumerable<HtmlSymbol> whitespace, IEnumerable<Htm
                 Span.ChunkGenerator = SpanChunkGenerator.Null; // The block chunk generator will render the prefix
                 Output(SpanKind.Markup);
 
-                // Read the values
-                while (!EndOfFile && !IsEndOfAttributeValue(quote, CurrentSymbol))
+                // Read the attribute value only if the value is quoted
+                // or if there is no whitespace between '=' and the unquoted value.
+                if (quote != HtmlSymbolType.Unknown || !whitespaceAfterEquals.Any())
                 {
-                    AttributeValue(quote);
+                    // Read the values
+                    while (!EndOfFile && !IsEndOfAttributeValue(quote, CurrentSymbol))
+                    {
+                        AttributeValue(quote);
+                    }
```

Comment:
"Values" for a single attribute i.e. name / value pair? Do you mean "symbols in the value"?